core/src/main/java/org/apache/druid/indexer/partitions/HashedPartitionsSpec.java

```diff
@@ -50,15 +51,18 @@ public HashedPartitionsSpec(
       @Nullable List<String> partitionDimensions
   )
   {
-    this(null, maxRowsPerSegment, numShards, partitionDimensions);
+    this(maxRowsPerSegment, numShards, partitionDimensions, null);
   }
 
   @JsonCreator
   public HashedPartitionsSpec(
-      @JsonProperty("targetPartitionSize") @Deprecated @Nullable Integer targetPartitionSize,
-      @JsonProperty("maxRowsPerSegment") @Nullable Integer maxRowsPerSegment,
+      @JsonProperty(PartitionsSpec.MAX_ROWS_PER_SEGMENT) @Nullable Integer maxRowsPerSegment,
```

Comment:
For consistency with the single dim spec, it seems like it would be better to call this `targetRowsPerSegment` to match `targetPartitionSize`, but that would mean deprecating yet another property

Reply:
hm, or it may be better to have a different rename for `maxPartitionSize` in the single dim spec, since `targetPartitionSIze` and `maxRowsPerSegment` were equivalent before

Reply:
You mean to rename "targetPartitionSize" in HashedPartitionsSpec to "targetRowsPerSegment"? My understanding is that hashed partitions do not have a "target" size as it is really a "max" size (which is why the former is deprecated).

@jihoonson described to me that when #8141 did some refactoring, the behavior of "targetPartitionSize" in SingleDimensionPartitionsSpec was wrongly made to match that of "maxRowsPerSegment". This PR separates the two so that single dimension partitioning can be made to honor a "target" size.

Reply:
Ah, got it, thanks for clarifying

Reply:
> You mean to rename "targetPartitionSize" in HashedPartitionsSpec to "targetRowsPerSegment"? My understanding is that hashed partitions do not have a "target" size as it is really a "max" size (which is why the former is deprecated).

Ah sorry, probably there was some misunderstanding. `targetPartitionSize` i n`HashedPartitionsSpec` (which I deprecated in favor of `maxRowsPerSegment`) is actually the target number of rows per segment. You can see how to compute `numShards` based on `targetPartitionSize` [here](https://github.com/apache/incubator-druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java#L633-L643). So, IMO, `targetRowsPerSegment` makes most sense for it too.

Reply:
Got it. I'll update `HashedPartitionsSpec` to deprecate `maxRowsPerSegment` and to add `targetRowsPerSegment`.